src/administration/components/AdminPanel.jsx

```diff
@@ -17,160 +12,227 @@ const electron = window.require('electron');
 const ipcRenderer = electron.ipcRenderer;
 
 class Admin extends Component {
-    state = {
-        volunteers: [],
-        inputFirstName: '',
-        inputLastName: '',
-        isSaveButtonEnabled: false
-    };
+    constructor(props) {
+        super(props);
+        this.state = {
+            volunteers: [],
+            inputFirstName: '',
+            inputLastName: '',
+            errorMessage: '',
+            firstNameValid: false,
+            lastNameValid: false
+        }
+        this.handleBlur = this.handleBlur.bind(this);
```

Comment:
You don't need to bind methods if they're declared as arrow functions. See [here](https://medium.com/@nikolalsvk/loosing-bind-this-in-react-8637ebf372cf).